website/src/components/UseAnalytics.tsx

```diff
@@ -0,0 +1,20 @@
+import ReactGA from 'react-ga4'
+
+type UseAnalyticsFunction = {
+  init: (trackingId: string) => void;
+  trackPageViewed: (path: string) => void;
+}
+
+export const useAnalytics = () : UseAnalyticsFunction => {
+  return {
+    init: (trackingId: string) => {
+      ReactGA.initialize(trackingId)
+    },
+    trackPageViewed: (path: string) => {
+      if (path) {
+        return ReactGA.pageview(path)
+      }
+      return ReactGA.pageview(window.location.pathname + window.location.search)
+    },
+  }
+}
```

Comment:
Let's have a unit test for this component, mock out `react-ga4` and check it propagates the right values